### ISSUE TITLE
Performance Issue Fine-Tuning D-FINE S with custom dataset

### ISSUE THREAD

**Issue by sergiosanchoasensio** [Peterande/D-FINE#108]
Hi everyone,

I’m fine-tuning the D-FINE S model initialized with Objects365 weights using a custom dataset, but I’m encountering significantly lower performance than expected.

**Performance Comparison**
In just a single epoch:
- D-FINE S achieves mAP@50=0.384.
- YOLOv11 S (initialized with COCO weights) achieves mAP@50=0.842.

**1. Dataset Preparation:**
  - Converted custom dataset annotations (x1,y1,x2,y2) to COCO format (x1, y1, w, h).
  - Since the dataset includes truncated images, I added in the train.py:
`ImageFile.LOAD_TRUNCATED_IMAGES = True`
  - This dataset contains only one class, category_id=0, so I set `num_classes: 1` and `remap_mscoco_category: False`.
 
**2. Configuration Changes:** 
  - Using dfine_hgnetv2_s_obj2custom.yml as the base configuration.
  - Modified only the dataset path and adjusted the batch size to 8 for both training and validation.
 
**3. Training Command:**
```
CUDA_VISIBLE_DEVICES=0 torchrun --master_port=7777 --nproc_per_node=1 train.py \
  -c configs/dfine/custom/objects365/dfine_hgnetv2_s_obj2custom.yml \
  --seed=0 -t dfine_s_obj365.pth
```

What could be causing the poor performance?
I’d appreciate any insights or suggestions for debugging this issue. Thanks in advance!

PS: I tried fine-tuning for 20 epochs, and the mAP@50 only improves slightly to 0.4. Using COCO weights instead of Objects365 weights does not lead to any improvement either.

**Comment by Peterande** [comment]
Can you try training from scratch without any pre-training weights?

**Comment by sergiosanchoasensio** [comment]
> Can you try training from scratch without any pre-training weights?

I trained the model from scratch but obtained an mAP@50 of -1.000. I suspect the issue may lie in the annotation format not aligning with the requirements of D-FINE. Could you clarify the correct format? Should the annotations follow COCO format (x1, y1, w, h), YOLO format (x_center, y_center, w, h), or another standard? Additionally, should the values be normalized or not?

Thank you in advance.

**Comment by prashant-dn** [comment]
Same here - I tried these 2 variants : 
`dfine_hgnetv2_${model}_obj365.yml` and `dfine_hgnetv2_${model}_custom.yml`
and the mAP50 never goes above 0.35 while even YOLOv9c gets to 0.7

**Comment by AnishMathewOommen** [comment]
@sergiosanchoasensio Could you please tell me how to use the pretrained model checkpoint file. Please share codes if you are no issues in sharing

**Comment by S130111** [comment]
I also encountered the same problem. Does anyone solve it?

**Comment by ArgoHA** [comment]
@AnishMathewOommen  I used this command for single GPU setup and custom dataset
`torchrun train.py -c configs/dfine/custom/objects365/dfine_hgnetv2_m_obj2custom.yml --use-amp --seed=0 -t dfine_m_coco.pth`

**Comment by FrancescoSaverioZuppichini** [comment]
I am also experiencing terrible performanes with dfine, it just sucks (more likely I suck at training) but I am using [this dataset](https://huggingface.co/datasets/Francesco/open-omniparser-dataset) with just one classes no matter what I do I end up with map at 0.15

**Comment by ArgoHA** [comment]
@FrancescoSaverioZuppichini try out [this repo](https://github.com/ArgoHA/custom_d_fine)

**Comment by FrancescoSaverioZuppichini** [comment]
> [@FrancescoSaverioZuppichini](https://github.com/FrancescoSaverioZuppichini) try out [this repo](https://github.com/ArgoHA/custom_d_fine)

Thanks a lot man, is on my todo list!

**Comment by FrancescoSaverioZuppichini** [comment]
> [@FrancescoSaverioZuppichini](https://github.com/FrancescoSaverioZuppichini) try out [this repo](https://github.com/ArgoHA/custom_d_fine)

man what the fuck your code is amazing, it is training so much faster and it is so well done - are you an advance AI from the future?
